roda-core/roda-core/src/main/java/org/roda/core/plugins/base/preservation/EditFileFormatPlugin.java

```diff
@@ -0,0 +1,302 @@
+  private void processFiles(IndexService index, ModelService model, StorageService storage, Report report,
+    JobPluginInfo jobPluginInfo, Job cachedJob, List<File> files) {
+    Report parametersReport = validateParameters();
+
+    for (File file : files) {
+      Report reportItem = PluginHelper.initPluginReportItem(this, file.getId(), File.class);
+      PluginHelper.updatePartialJobReport(this, model, reportItem, false, cachedJob);
+      if (parametersReport.getPluginState().equals(PluginState.SUCCESS)) {
+        if (file.isDirectory()) {
+          jobPluginInfo.incrementObjectsProcessedWithFailure();
+          reportItem.setPluginState(PluginState.SKIPPED).setPluginDetails("Skipping folder");
+        } else {
+          List<LinkingIdentifier> sources = new ArrayList<>();
+          try {
+            sources.add(setFileFormatMetadata(model, file, cachedJob.getId(), cachedJob.getUsername()));
+            jobPluginInfo.incrementObjectsProcessedWithSuccess();
+            reportItem.setPluginState(PluginState.SUCCESS);
+          } catch (RequestNotValidException | NotFoundException | AuthorizationDeniedException | GenericException
+            | PluginException e) {
+            LOGGER.error("Error setting format metadata on file {}: {}", file.getId(), e.getMessage(), e);
+
+            jobPluginInfo.incrementObjectsProcessedWithFailure();
+            reportItem.setPluginState(PluginState.FAILURE)
+              .setPluginDetails("Error setting format metadata on file " + file.getId() + ": " + e.getMessage());
+          }
+
+          try {
+            PluginHelper.createPluginEvent(this, file.getAipId(), file.getRepresentationId(), file.getPath(),
+              file.getId(), model, index, sources, null, reportItem.getPluginState(), "", true, cachedJob);
+          } catch (AuthorizationDeniedException | RequestNotValidException | NotFoundException | GenericException
+            | ValidationException | AlreadyExistsException e) {
+            LOGGER.error("Error creating event: {}", e.getMessage(), e);
+          }
+        }
+      }
+      else {
+        reportItem.setPluginState(PluginState.FAILURE);
+        reportItem.setPluginDetails(parametersReport.getPluginDetails());
+        jobPluginInfo.incrementObjectsProcessedWithFailure();
+      }
+
+      report.addReport(reportItem);
+      PluginHelper.updatePartialJobReport(this, model, reportItem, true, cachedJob);
+    }
+  }
+
+  private Report validateParameters() {
+    Report reportItem = PluginHelper.initPluginReportItem(this, null, File.class);
+    reportItem.setPluginState(PluginState.SUCCESS);
+    if (mimetype != null && !mimetype.isEmpty() && !mimetype.matches(RodaConstants.REGEX_MIME)) {
+      reportItem.setPluginState(PluginState.FAILURE).addPluginDetails(
+        "Invalid mimetype (expecting a string that conforms to \"" + RodaConstants.REGEX_MIME + "\".");
+    }
+    if (pronom != null && !pronom.isEmpty() && !pronom.matches(RodaConstants.REGEX_PUID)) {
```

Comment:
Use StringUtils from Apache Commons Lang. Use _notBlank_ or _blank_ methods to perform the null and empty validation

Reply:
https://github.com/keeps/roda/pull/3255/commits/de2e94ce3f2ffe9099904d69b8e10203caf3a2fb